### DISCUSSION TITLE
Hardware

### DISCUSSION THREAD

**Discussion by lorenzo-moni** [saul/demofile#273] [Questions] [answered]
What hardware should I use to host a web app that need to analyze demos at a reasonable high speed with multiples users? How many cores do I need to start a web app like csgostats? The download speed isn't a problem because all servers have a very high download speed but I was asking myself how much this parse process consumes in terms of CPU usage. Thank you in advance!

**Comment by petz15** [comment, marked as answer]
I am not answering out of experience from hosting a web service or writing a optimized parser but from parsing a bunch of demos from hltv.org: the parsing was usual not a big issue (atleast on a medium to fast desktop). I t took me roughly 30 seconds with an unoptimized code. My biggest issue was download and storage (I realize you wrote that internet speed is not an issue). So honestly I think most laptops from the past couple of years will be just fine for 2 demos per minute especially concerning that paraing is way faster than downloading. 

tl:dr couple of standard cpus will do the trick. storage and internet speed will be your bottlenecks
